File: 00DataService/SourceCode/LNMeterDataServiceSimulator/LNMeterDataServiceSimulator/frmLNSvrSimulator.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.CompilerServices;
using System.Security;
using System.Text;
using Microsoft.VisualBasic;
using Model;
using LNMeterInfrastructure.BLL;
using LNMeterInfrastructure.Common;
using System.Timers;
using ServiceMgr;
using System.Drawing;
using System.Windows.Forms;

public partial class frmLNSvrSimulator : Form
{
    private System.Timers.Timer _Cycletimer;
    private LNMeterCategory _lMeterCategory = null;

    public frmLNSvrSimulator()
    {
        InitializeComponent();
    }

    private void FireService()
    {
        try
        {
            AppendLogMsg("Service is starting.", (int)Variables.MsgCategoryFlag.MsgNormal);
            _lMeterCategory = new LNMeterCategory(LNGlobal.g_ServiceFilter);
            if (chkTimerControl.Checked)
            {
                _Cycletimer = new System.Timers.Timer(1);
                _Cycletimer.Elapsed += new ElapsedEventHandler(ProcessToTimerElapsed);
                _Cycletimer.Start();
            }
            else
            {
                LNSvrMgr.FireCollectDataAction(_lMeterCategory);
            }

            AppendLogMsg("Service is started successfully.", (int)Variables.MsgCategoryFlag.MsgNormal);
        }
        catch (Exception ex)
        {
            AppendLogMsg(ex.Message + Constants.vbCrLf + ex.StackTrace, (int)Variables.MsgCategoryFlag.MsgError);
        }
    }

    private void ProcessToTimerElapsed(object sender, ElapsedEventArgs e)
    {
        try
        {
            _Cycletimer.Interval = _lMeterCategory.CycleTime * 1000;
            //LNLogMgr.WriteLogAsy(string.Format("Service is running for the filter {0} with interval {1} seconds.", LNGlobal.g_ServiceFilter, _lMeterCategory.CycleTime.ToString()));
            AppendLogMsg(string.Format("Service is running for the filter {0} with interval {1} seconds.", LNGlobal.g_ServiceFilter, _lMeterCategory.CycleTime.ToString()), (int)Variables.MsgCategoryFlag.MsgNormal);
            LNSvrMgr.FireCollectDataAction(_lMeterCategory);
        }
        catch (Exception ex)
        {
            AppendLogMsg(ex.Message + Constants.vbCrLf + ex.StackTrace, (int)Variables.MsgCategoryFlag.MsgError);
        }
    }

    protected void StopService()
    {
        try
        {
            AppendLogMsg("Stopping the service.", (int)Variables.MsgCategoryFlag.MsgNormal);
            _Cycletimer.Stop();
            _Cycletimer.Dispose();
            AppendLogMsg("Service is stopped successfully.", (int)Variables.MsgCategoryFlag.MsgNormal);
        }
        catch (Exception ex)
        {
            AppendLogMsg(ex.Message + Constants.vbCrLf + ex.StackTrace, (int)Variables.MsgCategoryFlag.MsgError);
        }
    }

    private delegate void AppendLogMsgDel(string logMsg, Variables.MsgCategoryFlag MsgOption);
    static int lineNumber = 0;
    private void AppendLogMsg(string logMsg, int MsgOption)
    {
        try
        {
            logMsg = string.Format("{0} >>> {1}", DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss.fff"), logMsg);
            lineNumber += 1;
            rhLog.SelectionStart = 0;
            rhLog.SelectedText = lineNumber.ToString() + ". " + logMsg + Constants.vbCrLf + rhLog.SelectedText;
            rhLog.Select(0, rhLog.Lines[0].Length);
            switch (MsgOption)
            {
                case 0://MsgError
                    rhLog.SelectionColor = Color.Red;
                    break;
                case 1://MsgReminding
                    rhLog.SelectionColor = Color.Black;
                    break;
                case 2://MsgNormal
                    rhLog.SelectionColor = Color.Black;
                    break;
                case 3://MsgIncoming
                    rhLog.SelectionColor = Color.Blue;
                    break;
                case 4://MsgOutput
                    rhLog.SelectionColor = Color.ForestGreen;
                    break;
                default:
                    {
                        rhLog.SelectionColor = Color.Black;
                        break;
                    }
            }
        }
        catch (Exception ex)
        {
        }
    }

    private void cmdStartService_Click(object sender, EventArgs e)
    {
        try
        {
            FireService();
        }
        catch (Exception ex)
        {
            AppendLogMsg(ex.Message + Constants.vbCrLf + ex.StackTrace, (int)Variables.MsgCategoryFlag.MsgError);
        }
    }

    private void Button1_Click(object sender, EventArgs e)
    {
        try
        {

        }
        catch (Exception ex)
        {
            AppendLogMsg(ex.Message + Constants.vbCrLf + ex.StackTrace, (int)Variables.MsgCategoryFlag.MsgError);
        }
    }

    private void cmdApply_Click(object sender, EventArgs e)
    {
        try
        {
            float lnA = IEEE754Helper.HexToFloat(this.txtOrgHex.Text.Replace(" ", ""));
            AppendLogMsg(string.Format("Hex to Float:Hex[{1}] --> Float[{0}]", lnA.ToString(), this.txtOrgHex.Text), (int)Variables.MsgCategoryFlag.MsgNormal);
        }
        catch (Exception ex)
        {
            AppendLogMsg(ex.Message + Constants.vbCrLf + ex.StackTrace, (int)Variables.MsgCategoryFlag.MsgError);
        }
    }

    private void Button1_Click_1(object sender, EventArgs e)
    {
        try
        {
            string lnCommand = LNSvrMgr.ToGenCommand(this.txtAddr.Text, this.txtCommand.Text);
            AppendLogMsg(string.Format("Hex Command: [{0}]", lnCommand), (int)Variables.MsgCategoryFlag.MsgNormal);
        }
        catch (Exception ex)
        {
            AppendLogMsg(ex.Message + Constants.vbCrLf + ex.StackTrace, (int)Variables.MsgCategoryFlag.MsgError);
        }
    }
}

File: DMNB17LNMeterDataService/00DataService/SourceCode/LNMeterDataCollectionSystem/Form/frmLNSvrSimulator.vb
Imports Model
Imports LNMeterInfrastructure.BLL
Imports LNMeterInfrastructure.Common
Imports System
Imports System.Timers
Imports LNMeterDataCollectionSystem.ServiceMgr

Public Class frmLNSvrSimulator
    Private _Cycletimer As Timer
    Private _lMeterCategory As LNMeterCategory = Nothing

    Private Sub FireService()
        Try
            AppendLogMsg("Service is starting.", MsgCategoryFlag.MsgNormal)
            _lMeterCategory = New LNMeterCategory(LNGlobal.g_ServiceFilter) '
            'LNSvrMgr.FireCollectDataAction(_lMeterCategory)

            _Cycletimer = New Timer(_lMeterCategory.CycleTime * 1000)
            AddHandler _Cycletimer.Elapsed, New ElapsedEventHandler(AddressOf ProcessToTimerElapsed)
            _Cycletimer.Start()
            AppendLogMsg("Service is started successfully.", MsgCategoryFlag.MsgNormal)
        Catch ex As Exception
            AppendLogMsg(ex.Message & vbCrLf & ex.StackTrace, MsgCategoryFlag.MsgError)
        End Try
    End Sub

    Private Sub ProcessToTimerElapsed(ByVal sender As Object, ByVal e As ElapsedEventArgs)
        Try
            AppendLogMsg(String.Format("Service is running for the filter {0}.", LNGlobal.g_ServiceFilter), MsgCategoryFlag.MsgNormal)
            LNSvrMgr.FireCollectDataAction(_lMeterCategory)
        Catch ex As Exception
            AppendLogMsg(ex.Message & vbCrLf & ex.StackTrace, MsgCategoryFlag.MsgError)
        End Try
    End Sub

    Protected Sub StopService()
        Try
            AppendLogMsg("Stopping the service.", MsgCategoryFlag.MsgNormal)
            _Cycletimer.[Stop]()
            _Cycletimer.Dispose()
            AppendLogMsg("Service is stopped successfully.", MsgCategoryFlag.MsgNormal)
        Catch ex As Exception
            AppendLogMsg(ex.Message & vbCrLf & ex.StackTrace, MsgCategoryFlag.MsgError)
        End Try
    End Sub

    Private Delegate Sub AppendLogMsgDel(ByVal logMsg As String, ByVal MsgOption As MsgCategoryFlag)
    Private Sub AppendLogMsg(ByVal logMsg As String, ByVal MsgOption As MsgCategoryFlag)
        Try
            Static lineNumber As Integer = 0
            If Me.InvokeRequired Then
                Me.rhLog.BeginInvoke(New AppendLogMsgDel(AddressOf AppendLogMsg), logMsg, MsgOption)
                Return
            End If

            logMsg = String.Format("{0} >>> {1}", Now().ToString("yyyy-MM-dd HH:mm:ss.fff"), logMsg)
            lineNumber += 1
            rhLog.SelectionStart = 0
            rhLog.SelectedText = lineNumber.ToString & ". " & logMsg + vbCrLf + rhLog.SelectedText
            rhLog.Select(0, rhLog.Lines(0).Length)

            Select Case MsgOption
                Case MsgCategoryFlag.MsgError
                    rhLog.SelectionColor = Color.Red
                Case MsgCategoryFlag.MsgNormal
                    rhLog.SelectionColor = Color.Black
                Case MsgCategoryFlag.MsgReminding
                    rhLog.SelectionColor = Color.Black
                Case MsgCategoryFlag.MsgOutput
                    rhLog.SelectionColor = Color.ForestGreen
                Case MsgCategoryFlag.MsgIncoming
                    rhLog.SelectionColor = Color.Blue
                Case Else
                    rhLog.SelectionColor = Color.Black
            End Select
        Catch ex As Exception

        End Try
    End Sub

    Private Sub cmdOK_Click(sender As Object, e As EventArgs) Handles cmdOK.Click
        Try
            FireService()

        Catch ex As Exception
            AppendLogMsg(ex.Message & vbCrLf & ex.StackTrace, MsgCategoryFlag.MsgError)
        End Try
    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles cmdHostSimulator.Click
        Try
            Dim f As Form = New frmSPSSimulator()
            f.ShowDialog()
        Catch ex As Exception
            AppendLogMsg(ex.Message & vbCrLf & ex.StackTrace, MsgCategoryFlag.MsgError)
        End Try
    End Sub

    Private Sub cmdApply_Click(sender As Object, e As EventArgs) Handles cmdApply.Click
        Try
            Dim lnA As Single = IEEE754Helper.HexToFloat(Me.txtOrgHex.Text)
            AppendLogMsg(String.Format("Hex to Float:Hex[{1}] --> Float[{0}]", lnA.ToString(), Me.txtOrgHex.Text), MsgCategoryFlag.MsgNormal)
        Catch ex As Exception
            AppendLogMsg(ex.Message & vbCrLf & ex.StackTrace, MsgCategoryFlag.MsgError)
        End Try
    End Sub

    Private Sub Button1_Click_1(sender As Object, e As EventArgs) Handles Button1.Click
        Try
            Dim lnCommand As String = LNSvrMgr.ToGenCommand(Me.txtAddr.Text, Me.txtCommand.Text)
            AppendLogMsg(String.Format("Hex Command: [{0}]", lnCommand), MsgCategoryFlag.MsgNormal)
        Catch ex As Exception
            AppendLogMsg(ex.Message & vbCrLf & ex.StackTrace, MsgCategoryFlag.MsgError)
        End Try
    End Sub
End Class
